pygen/pygen_src/isa/riscv_instr.py

```diff
@@ -0,0 +1,341 @@
+                continue
+            if (("SP" in cfg.reserved_regs) and (instr_name == "C_ADDI16SP")):
+                continue
+            if (cfg.enable_sfence and instr_name == "SFENCE_VMA"):
+                continue
+            if (instr_name in ["FENCE","FENCE_I","SFENCE_VMA"]):
+                continue
+            if (instr_inst.group in self.supported_isa):
+                self.instr_category[instr_inst.category.name].append(instr_name)
+                self.instr_group[instr_inst.group.name].append(instr_name)
+                self.instr_names.append(instr_name)
+                
+        self.build_basic_instruction_list(cfg)
+        self.create_csr_filter(cfg)
+        
+    def create_instr(self,instr_name):
+        instr_inst = eval("rv32i_instr.riscv_"+instr_name+"_instr()")
+        if(instr_inst == None):
+            logging.critical("Failed to create instr: %0s", instr_name)
+        return instr_inst
+
+    def is_supported(self, cfg):
+        return 1
+
+    def build_basic_instruction_list(self, cfg):
+        self.basic_instr = (self.instr_category["SHIFT"] + self.instr_category["ARITHMETIC"] + self.instr_category["LOGICAL"] + self.instr_category["COMPARE"])
+        if(cfg.no_ebreak):
+            self.basic_instr.append("EBREAK")
+            for items in self.supported_isa:
+                if("RV32C" in self.supported_isa and not(cfg.disable_compressed_instr)):
+                    self.basic_instr.append("C_EBREAK")
+                    break
+        if(cfg.no_dret==0):
+            slef.basic_instr.append("DRET")
+        if(cfg.no_fence==0):
+            self.basic_instr.append(self.instr_category["SYNCH"])
+        if(cfg.no_csr_instr == 0 and cfg.init_privileged_mode == "MACHINE_MODE"):
+            self.basic_instr.append(self.instr_category["CSR"])
+        if(cfg.no_wfi == 0):
+            self.basic_instr.append("WFI")
+
+    def create_csr_filter(self, cfg):
+        self.include_reg.clear()
+        self.exclude_reg.clear()
+
+        if(cfg.enable_illegal_csr_instruction):
+            self.exclude_reg = self.implemented_csr
+        elif(cfg.enable_access_invalid_csr_level):
+            self.include_reg = cfg.invalid_priv_mode_csrs
+        else:
+            if(cfg.init_privileged_mode == "MACHINE_MODE"):    # Machine Mode
+                self.include_reg.append("MSCRATCH")
+            elif(cfg.init_privileged_mode=="SUPERVISOR_MODE"): # Supervisor Mode
+                self.include_reg.append("SSCRATCH")
+            else:                                              # User Mode
+                self.include_reg.append("USCRATCH") 
+
+    def get_rand_instr(self):
+        pass #TODO
+
+    def get_load_store_instr(self, load_store_instr):
+        instr_h = riscv_instr()
+        if(len(load_store_instr) == 0):
+            load_store_instr = self.instr_category["LOAD"] + self.instr_category["STORE"]
+        self.idx = random.randrange(0,len(load_store_instr)-1)    
```

Comment:
nit: `random.randrange(0, len(load_store_instr) - 1)`

Reply:
Should a CI check for `flake8`, and `linting`?